File: adventofcode/java/year2016/day08/Day08Test.java
package adventofcode.java.year2016.day08;

import java.io.BufferedInputStream;
import java.io.FileInputStream;
import java.io.FileReader;
import java.io.IOException;

import org.junit.Test;
import org.junit.Assert;

public class Day08Test {
    @Test
    public void part1Actual() throws IOException {
        try (var input = new FileReader("adventofcode/inputs/2016/08")) {
            var output = "116";
            var result = Day08.part1(input);
            Assert.assertEquals("no error", "", result.error);
            Assert.assertEquals("correct output", output, result.answer);
        }
    }

     @Test
     public void part2Actual() throws IOException {
         try (var input = new FileReader("adventofcode/inputs/2016/08");
              var outputStream = new FileInputStream("adventofcode/java/testdata/year2016/day08/output")) {
             var outputBytes = new BufferedInputStream(outputStream).readAllBytes();
             var output = new String(outputBytes);
             var result = Day08.part2(input);
             Assert.assertEquals("no error", "", result.error);
             Assert.assertEquals("correct output", output, result.answer);
         }
     }
}

File: adventofcode/cpp/year2019/day08/day08_test.cc
#include "adventofcode/cpp/year2019/day08/day08.h"

#include <istream>
#include <fstream>
#include <string>

#include "gtest/gtest.h"

#include "adventofcode/cpp/adventofcode.h"

TEST(Year2019Day08, Part1Actual) {
  std::ifstream input("adventofcode/data/year2019/day08/actual.in");
  std::string output = "2032";
  adventofcode::answer_t a = day08::part1(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
  input.close();
}

TEST(Year2019Day08, Part2Actual) {
  std::ifstream input("adventofcode/data/year2019/day08/actual.in");
  std::ifstream output_file("adventofcode/cpp/year2019/day08/testdata/p2out");
  std::string output(std::istreambuf_iterator<char>(output_file), {});
  adventofcode::answer_t a = day08::part2(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
  input.close();
}
